core/api/src/app/wallets/update-pending-invoices.ts

```diff
@@ -302,6 +316,7 @@ const updatePendingInvoiceBeforeFinally = async ({
       "invoices.finalRecipient": JSON.stringify(recipientWalletDescriptor),
     })
 
+    // Is this crossing the lock boundary?
```

Comment:
please remove, also not sure... what you mean? this can be done outside the lock but need to be done previous to ledger record (that must be inside lock)

Reply:
I will remove it, but `lnInvoiceLookup` is populated before the lock, so its state could change by the time the lock is acquired.

Reply:
ahh I thought you referred to settle... you are right, feel free to move it